install_config/advanced_ldap_configuration/sssd_for_ldap_failover.adoc

```diff
@@ -0,0 +1,425 @@
+
+. Generate the API client certificate that the authenticating proxy will use
+to prove its identity to {product-title}. This is necessary and prevents
+malicious users from impersonating the proxy and sending fake identities.
+
+. Create a new CA to sign this client certificate:
++
+====
+----
+# oadm ca create-signer-cert \
+  --cert='/etc/origin/proxy/proxyca.crt' \
+  --key='/etc/origin/proxy/proxyca.key' \
+  --name='openshift-proxy-signer@`date +%s`' \
+  --serial='/etc/origin/proxy/proxyca.serial.txt'
+----
+====
++
+`date +%s` makes the signer unique. You can use any name you prefer, however.
++
+====
+----
+# oadm create-api-client-config \
+    --certificate-authority='/etc/origin/proxy/proxyca.crt' \
+    --client-dir='/etc/origin/proxy' \
+    --signer-cert='/etc/origin/proxy/proxyca.crt' \
+    --signer-key='/etc/origin/proxy/proxyca.key' \
+    --signer-serial='/etc/origin/proxy/proxyca.serial.txt' \
+    --user='system:proxy'
+# cat /etc/origin/proxy/system\:proxy.crt \
+      /etc/origin/proxy/system\:proxy.key \
+      > /etc/origin/proxy/authproxy.pem
+----
+====
+
+[[sssd-phase-2-authenticating-proxy-setup]]
+== Phase 2: Authenticating Proxy Setup
+
+This section guides you through the steps to authenticate the proxy setup.
+
+[[phase-2-step-1-copy-certificates]]
+=== Step 1: Copy Certificates
+
+From *_openshift.example.com_*, securely copy the necessary certificates to the
+proxy machine:
+
+====
+----
+# scp /etc/origin/proxy/master/ca.crt \
+      root@proxy.example.com:/etc/pki/CA/certs/
+
+# scp /etc/origin/proxy/proxy.example.com.crt \
+      /etc/origin/proxy/authproxy.pem \
+      root@proxy.example.com:/etc/pki/tls/certs/
+
+# scp /etc/origin/proxy/proxy.example.com.key \
+      root@proxy.example.com:/etc/pki/tls/private/
+----
+====
+
+[[phase-2-step-2-sssd-configuration]]
+=== Step 2: SSSD Configuration
+. Install a new VM with a recent operating system. Run SSSD 1.12.0 or later so
+that you can use the *mod_identity_lookup* module. The examples in this topic
+use a Red Hat Enterprise Linux 7.2 Server.
+
+. Install all of the necessary dependencies:
++
+====
+----
+# yum install -y sssd \
+                 sssd-dbus \
+                 realmd \
+                 httpd \
+                 mod_session \
+                 mod_ssl \
+                 mod_lookup_identity \
+                 mod_authnz_pam
+----
+====
++
+This gives you the needed SSSD and the web server components.
+
+. Set up SSSD to authenticate this VM against the LDAP server. If the LDAP server
+is a FreeIPA or Active Directory environment, then *realmd* can be used to join
+this machine to the domain.
++
+====
+----
+# realm join ldap.example.com
+----
+====
++
+If you are not running a domain, then use the *authconfig* tool:
++
+====
+----
+# authconfig --update --enablesssd --enablesssdauth \
+             --ldapserver=ldap.example.com \
+             --enableldaptls \
+             --ldaploadcert=http://ldap.example.com/ca.crt
+----
+====
```

Comment:
It might be better to just redirect to Red Hat's official documentation on joining a domain with SSSD, rather than trying to reimplement it here. (Among other things, authconfig in 7.2 seems to be a little flaky and doesn't always work right, so I wouldn't want to recommend it).

I would keep the `realm join ldap.example.com` piece for FreeIPA and AD, because that's going to be the common case anyway (I expect) and just link to https://access.redhat.com/documentation/en-US/Red_Hat_Enterprise_Linux/7/html/System-Level_Authentication_Guide/authconfig-ldap.html for more advanced cases.